semantic_release/hvcs/github.py

```diff
@@ -186,7 +186,10 @@ def asset_upload_url(self, release_id: str) -> str:
         :return: URL found to upload for a release
         """
         # https://docs.github.com/en/enterprise-server@3.5/rest/releases/assets#upload-a-release-asset ?
-        return f"{self.api_url}/repos/{self.owner}/{self.repo_name}/releases/{release_id}/assets"
+        response = self.session.get(
+            f"{self.api_url}/repos/{self.owner}/{self.repo_name}/releases/{release_id}",
+        )
```

Comment:
This introduces a new source of errors that didn't exist before - what if you get a 404?